### PR TITLE
Document OpenAPI schema for AWSPerformanceInsightsSource

### PR DESCRIPTION
**Pull request by antoineco**
Part of #313

---

Demo :

(just printing the top level spec, please check locally for yourself if you're interested in printing the doc for sub-attributes)

```console
$ kubectl explain awsperformanceinsightssources.spec
KIND:     AWSPerformanceInsightsSource
VERSION:  sources.triggermesh.io/v1alpha1

RESOURCE: spec <Object>

DESCRIPTION:
     Desired state of the event source.

FIELDS:
   arn  <string> -required-
     ARN of the Amazon RDS database instance to receive metrics for. The
     expected format is documented at
     https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonrds.html#amazonrds-resources-for-iam-policies.

   credentials  <Object>
     Credentials to interact with the Amazon RDS and Performance Insights APIs.
     For more information about AWS security credentials, please refer to the
     AWS General Reference at
     https://docs.aws.amazon.com/general/latest/gr/aws-security-credentials.html

   metricQueries        <[]string> -required-
     List of queries that determine what metrics will be sourced from Amazon
     Performance Insights. Each item represents the 'metric' attribute of a
     MetricQuery. For more information, please refer to the Performance Insights
     API reference at
     https://docs.aws.amazon.com/performance-insights/latest/APIReference/API_MetricQuery.html

   pollingInterval      <string> -required-
     Duration which defines how often metrics should be pulled from Amazon
     Performance Insights. Expressed as a duration string, which format is
     documented at https://pkg.go.dev/time#ParseDuration.

   sink <Object> -required-
     The destination of events generated from Amazon Performance Insights metrics.
```

### CODE REVIEW

File: config/300-awsperformanceinsights.yaml [hunk 1]
```diff
@@ -98,12 +117,14 @@ spec:
                         - name
                         - key
                     oneOf:
-                    - required: ['value']
-                    - required: ['valueFromSecret']
+                    - required: [value]
+                    - required: [valueFromSecret]
               sink:
+                description: The destination of events generated from Amazon Performance Insights metrics.
```
Comment (JeffNeff) [inline config/300-awsperformanceinsights.yaml:123]:
```suggestion
                description: The destination of events generated from the Amazon Performance Insights Source.
```

Comment (antoineco) [reply config/300-awsperformanceinsights.yaml:123]: What I wanted to express is that events are generated from metrics, if that makes sense.
An event is generated based on the observed metrics.

How would you express that better?

File: pkg/apis/sources/v1alpha1/awsperformanceinsights_types.go [hunk 2]
```diff
@@ -48,13 +48,24 @@ var (
...
+	// Expressed as a duration string, which format is documented at https://pkg.go.dev/time#ParseDuration.
+	PollingInterval apis.Duration `json:"pollingInterval"`
 
+	// List of queries that determine what metrics will be sourced from Amazon Performance Insights.
+	//
+	// Each item represents the 'metric' attribute of a MetricQuery.
+	// https://docs.aws.amazon.com/performance-insights/latest/APIReference/API_MetricQuery.html
+	//
+	// +optional
 	MetricQueries []string `json:"metricQueries"`
```
Comment (JeffNeff) [inline pkg/apis/sources/v1alpha1/awsperformanceinsights_types.go:65]: are we moving this to optional? do we not need to update the reconcile for that here -> https://github.com/triggermesh/aws-event-sources/blob/fcaa5a70e3cec4e67cfd7fa676be1308ca65777d/pkg/reconciler/awsperformanceinsightssource/adapter.go#L61 @antoineco

Comment (antoineco) [reply pkg/apis/sources/v1alpha1/awsperformanceinsights_types.go:65]: Oh, right, that wasn't intentional, I did a bad copy-paste from the CloudWatch source!

Comment (JeffNeff) [reply pkg/apis/sources/v1alpha1/awsperformanceinsights_types.go:65]: I was a little confused :)

File: config/300-awsperformanceinsights.yaml (hunk 1 above)
Comment (JeffNeff) [reply config/300-awsperformanceinsights.yaml:123]: @cab105  we need some proper English guidance xd

File: pkg/apis/sources/v1alpha1/awsperformanceinsights_types.go (hunk 2 above)
Comment (antoineco) [reply pkg/apis/sources/v1alpha1/awsperformanceinsights_types.go:63]: Fixed in [fcaa5a70..0e77c19d](https://github.com/triggermesh/aws-event-sources/compare/fcaa5a70e3cec4e67cfd7fa676be1308ca65777d..0e77c19d78c9e9135c6a2d9667a521fddad2cabd)

File: config/300-awsperformanceinsights.yaml (hunk 1 above)
Comment (JeffNeff) [reply config/300-awsperformanceinsights.yaml:123]: this is more of a nit. im fine with it as is. it just sounds a bit off to me personally. don't mean it as a blocker we can push past this. ill approve

Comment (antoineco) [reply config/300-awsperformanceinsights.yaml:123]: Would that sound better?

"The destination of events generated based on Amazon Performance Insights metrics."

**Review by JeffNeff — APPROVED** [review]
lgtm

File: config/300-awsperformanceinsights.yaml (hunk 1 above)
Comment (JeffNeff) [reply config/300-awsperformanceinsights.yaml:123]: lets just remove metrics and go with `The destination of events generated by Amazon Performance Insights` ?

Comment (antoineco) [reply config/300-awsperformanceinsights.yaml:123]: You got it

Comment (antoineco) [reply config/300-awsperformanceinsights.yaml]: Fixed in [0e77c19d..81de3691](https://github.com/triggermesh/aws-event-sources/compare/0e77c19d78c9e9135c6a2d9667a521fddad2cabd..81de3691bd53e353ebe5c3f6f078e2c618106a66)

**Review by cab105 — APPROVED** [review]

**Pull request merged**